client/signup/steps/about/index.jsx

```diff
@@ -267,13 +267,16 @@ class AboutStep extends Component {
 		}
 
 		//Site Topic
+		const sanitizedSiteTopicInput =
+			findKey( hints, partial( isEqual, siteTopicInput ) ) || siteTopicInput;
```

Comment:
Does `partial( isEqual, siteTopicInput )` the same thing as `( siteTopic ) => siteTopic == siteTopicInput` (i.e. `function (siteTopic ) { return siteTopic == siteTopicInput; }`)? The name `partial` implies partial matches but `isEqual` contradicts that. I'd prefer a more clear function.

Reply:
Also, I'd suggest to use `englishSiteTopic` instead of `sanitizedSiteTopicInput`. Apart from this being more clear about the variable contents, you're not sanitizing the input (which implies stripping away invalid characters etc) but actually looking up the value in the `hints` object.

Reply:
> Does partial( isEqual, siteTopicInput ) the same thing as ( siteTopic ) => siteTopic == siteTopicInput (i.e. function (siteTopic ) { return siteTopic == siteTopicInput; })? The name partial implies partial matches but isEqual contradicts that. I'd prefer a more clear function.

This was a code snippet I found online but I like yours better because it is more clear. 

> Also, I'd suggest to use englishSiteTopic instead of sanitizedSiteTopicInput. Apart from this being more clear about the variable contents, you're not sanitizing the input (which implies stripping away invalid characters etc) but actually looking up the value in the hints object.

Good suggestion. I'll update the name.